### PR TITLE
Import network isolation and MetalLB patches

### PR DESCRIPTION
**Pull request by fmount**
Support attaching pods to networks using net-attach-defs and exposing api via metallb.

### CODE REVIEW

**Comment by fmount** [comment]
@stuggi Stil WIP but I imported the same changes on a diff branch

File: controllers/manilashare_controller.go [hunk 1]
```diff
@@ -384,8 +419,9 @@ func (r *ManilaShareReconciler) reconcileNormal(ctx context.Context, instance *m
 	//
 
 	// Deploy a statefulset
+	ssDef := manilashare.StatefulSet(instance, inputHash, serviceLabels, serviceAnnotations)
 	ss := statefulset.NewStatefulSet(
-		manilashare.StatefulSet(instance, inputHash, serviceLabels),
+		ssDef,
```
Comment (stuggi) [inline controllers/manilashare_controller.go:424]: same could keep the `manilashare.StatefulSet(instance, inputHash, serviceLabels, serviceAnnotations)` call here.

**Review by stuggi — COMMENTED** [review]
thanks @fmount looks good! just two minor comments.

File: controllers/manilascheduler_controller.go [hunk 2]
```diff
@@ -383,8 +419,9 @@ func (r *ManilaSchedulerReconciler) reconcileNormal(ctx context.Context, instanc
 	//
 
 	// Deploy a statefulset
+	ssDef := manilascheduler.StatefulSet(instance, inputHash, serviceLabels, serviceAnnotations)
 	ss := statefulset.NewStatefulSet(
-		manilascheduler.StatefulSet(instance, inputHash, serviceLabels),
+		ssDef,
```
[the remark this answers is not in the document]
Comment (fmount) [reply controllers/manilascheduler_controller.go:424]: Makes sense, it's not used in any other place, so we can restore the definition within the statefulset. Doing it in my next iteration.

File: controllers/manilashare_controller.go (hunk 1 above)
Comment (fmount) [reply controllers/manilashare_controller.go:424]: +1

File: api/v1beta1/manilaapi_types.go [hunk 3]
```diff
@@ -103,6 +111,9 @@ type ManilaAPIStatus struct {
 
 	// ServiceIDs
 	ServiceIDs map[string]string `json:"serviceIDs,omitempty"`
+
+	// NetworkAttachments status of the deployment pods
+	NetworkAttachments map[string][]string `json:"networkAttachments,omitempty"`
 }
 
 //+kubebuilder:object:root=true
```
[the remark this answers is not in the document]
Comment (fmount) [reply api/v1beta1/manilaapi_types.go:119]: Yes! Good catch, thanks, adding in my next push!

**Comment by fmount** [comment]
@stuggi this should be now ready, let me know if you think we need additional changes!

**Review by stuggi — APPROVED** [review]
/lgtm

**Comment by viroel** [comment]
/test all

lets see if Prow can trigger jobs here

**Comment by gouthampacha** [comment]
/lgtm
/approve 

perfect, thank you @fmount and @viroel for triggering the CI

**Review by gouthampacha — APPROVED** [review]

**Pull request merged**